portal/server/generateApp.ts

```diff
@@ -61,7 +66,11 @@ const generateApp = () => {
     };
   }
 
-  const redisClient = redis.createClient(process.env.REDIS_PORT, process.env.REDIS_HOSTNAME, redisOptions);
+  const redisClient = redis.createClient(
+    parseInt(process.env.REDIS_PORT ?? '6379', 10),
```

Comment:
TS doesn't like `REDIS_PORT`  being undefined here.
Should I add more error handling for missing/invalid environment variables?

Reply:
In E2ETF, we introduce a helper function called `asString` which uses a type predicate telling typescript to treat the value as a string and throws if the value is not a string:
```
/**
 * @param value - the value to check
 * @param context - provides context in the error message if 'value' is not a
 *  string. Usually would be the name of the variable being passed in
 */
export const asString = (value: unknown, context: string): string => {
  if (!isString(value)) {
    throw new Error(`Expected ${context} to be a string, but was ${typeof value}`);
  }

  return value as string;
};
```

This comes from https://github.com/UK-Export-Finance/dtfs2/blob/main-stbr-tfm-5/trade-finance-manager-api/src/utils/validation.ts

Reply:
Alternatively, just use `process.env.REDIS_PORT || '6379'` instead of `??`